qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -131,82 +132,37 @@ def __getitem__(self, item: Union[int, str]):
             return self._x[item]
         if isinstance(item, str):
             return self._variables_dict[item]
-        raise QiskitOptimizationError("Integer or string parameter required, instead "
-                                      + type(item).__name__ + " provided.")
+        raise QiskitOptimizationError(
+            "Integer or string parameter required, instead '{}' provided.".format(item))
 
     @property
-    def variables_dict(self) -> Optional[Dict[str, int]]:
-        """Returns the pairs of variable names and values under optimization.
-
-        Returns:
-            The pairs of variable names and values under optimization.
-        """
-        return self._variables_dict
-
-    @property
-    def variable_names(self) -> Optional[List[str]]:
-        """Returns the list of variable names under optimization.
-
-        Returns:
-            The list of variable names under optimization.
-        """
-        return self._variable_names
-
-    @property
-    def x(self) -> Any:
+    def x(self) -> Union[List[float], np.ndarray]:
         """Returns the optimal value found in the optimization.
 
         Returns:
             The optimal value found in the optimization.
         """
         return self._x
 
-    @x.setter  # type: ignore
-    def x(self, x: Any) -> None:
-        """Set a new optimal value.
-
-        Args:
-            x: The new optimal value.
-        """
-        self._x = x
-
     @property
-    def fval(self) -> Any:
+    def fval(self) -> float:
         """Returns the optimal function value.
 
         Returns:
             The function value corresponding to the optimal value found in the optimization.
         """
         return self._fval
 
-    @fval.setter  # type: ignore
-    def fval(self, fval: Any) -> None:
-        """Set a new optimal function value.
-
-        Args:
-            fval: The new optimal function value.
-        """
-        self._fval = fval
-
     @property
-    def results(self) -> Any:
+    def raw_results(self) -> Optional[Any]:
```

Comment:
I think this can just be 'Any' since Any can include any type, which includes NoneType (the type of None)

Reply:
Thanks. I will fix it.